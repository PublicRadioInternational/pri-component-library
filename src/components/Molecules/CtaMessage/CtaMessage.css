/* import colors... */
@value colors: "../../00_global/colors.css";
@value gray, orange, grayLighter from colors;

/* import breakpoints */
@value medium as bp-medium from "../../00_global/breakpoints.css";

@define-mixin prompt {
  display: block;
  position: relative;
  box-sizing: border-box;
  background-color: gray;
  color: grayLighter;
  text-align: center;
}

@define-mixin loadUnderBtn {
  padding: 0 0.27rem;
  font-size: 0.8rem;
  line-height: 1.6rem;
  white-space: nowrap;
}

.pushDown {
  @mixin prompt;

  display: none;
}

.loadUnder {
  @mixin prompt;

  position: fixed;
  bottom: 0;
  width: 100%;

  padding: 4px 42px;
  border-top: 2px solid orange;
}

.close {
  display: block;
  position: absolute;
  top: 0;
  right: 0;

  padding: 10px;
  background: none;
  border: none;

  color: currentColor;

  cursor: pointer;
}

.close:hover {
  color: orange;
}

.logo {
  margin-bottom: 1rem;
}

.title {
  margin-bottom: 1rem;
  color: inherit;
  font-size: 1.5rem;
}

.description {
  font-size: 1.1rem;
  line-height: 1.4rem;
}

.actions * + * {
  margin-left: 1rem;
}

.loadUnder .content > * {
  display: inline-block;
  margin-bottom: 0;
}

.loadUnder .content > * + * {
  margin-left: 0.5rem;
}

.loadUnder .title {
  font-size: 1rem;
}

.loadUnder .close {
  top: 50%;
  transform: translateY(-50%);
  padding: 12px;
}

.loadUnder .logo,
.loadUnder .close {
  display: none;
}

.loadUnder .logo + * {
  margin-left: 0;
}

.loadUnder .actions {
  display: block;
  margin: 10px auto;
}

.loadUnder .actions > * {
  margin: 0.5rem 0.5rem;
}

.loadUnder .actionBtn,
.loadUnder .dismissBtn {
  @mixin loadUnderBtn;
}

@media bp-medium {
  .pushDown {
    display: block;
    padding: 35px 55px;
  }

  .loadUnder .logo {
    display: inline-block;
    vertical-align: middle;
    margin-right: 0.5em;
  }

  .loadUnder .actions {
    display: inline-block;
    margin: 0 0 0 0.5rem;
  }

  .loadUnder .dismissBtn {
    display: none;
  }

  .loadUnder .close {
    display: block;
  }
}
