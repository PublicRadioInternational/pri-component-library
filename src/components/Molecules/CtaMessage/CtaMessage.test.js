/**
 * @file CtaMessage.test.js
 * Contains tests for CtaMessage.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import CtaMessage from './CtaMessage.component';

describe('<CtaMessage />', () => {
  it('Matches the CTA Message Default snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd'
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the CTA Message Logo snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            showLogo: true
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Logo is small for loadUnder', () => {
    const wrapper = mount(
      <CtaMessage
        type="loadUnder"
        data={{
          name: 'test_1',
          hash: '1234abcd',
          showLogo: true
        }}
      />
    );
    expect(wrapper.find('.logo svg').prop('width')).toEqual(90);
  });

  it('Matches the CTA Message Title snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            title: 'Test #1'
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the CTA Message Title snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            title: 'Test title'
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the CTA Message Decription snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            description: 'Test description...'
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the CTA Message Action button snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            action: {
              label: 'Do it',
              url: 'https://www.google.com/'
            }
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the CTA Message Blue Action button snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            action: {
              label: 'Do it',
              url: 'https://www.google.com/',
              btnColor: 'Blue'
            }
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the CTA Message Dismiss button snapshot', () => {
    const component = renderer
      .create(
        <CtaMessage
          data={{
            name: 'test_1',
            hash: '1234abcd',
            action: {
              label: 'Do it',
              url: 'https://www.google.com/'
            },
            dismiss: {
              label: "Don't do it"
            }
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('onPromptClose handler is called when close is click', () => {
    const actionFn = jest.fn();
    const wrapper = mount(
      <CtaMessage
        onClose={actionFn}
        data={{
          name: 'test_1',
          hash: '1234abcd',
          action: {
            label: 'Do it',
            url: 'https://www.google.com/'
          }
        }}
      />
    );
    wrapper.find('button.close').simulate('click');
    expect(actionFn).toHaveBeenCalled();
  });

  it('onPromptClose handler is called when action button is click', () => {
    const actionFn = jest.fn();
    const wrapper = mount(
      <CtaMessage
        onClose={actionFn}
        data={{
          name: 'test_1',
          hash: '1234abcd',
          action: {
            label: 'Do it',
            url: 'https://www.google.com/'
          }
        }}
      />
    );
    wrapper.find('a.actionBtn').simulate('click');
    expect(actionFn).toHaveBeenCalled();
  });

  it('onPromptClose handler is called when dismiss button is click', () => {
    const actionFn = jest.fn();
    const wrapper = mount(
      <CtaMessage
        onClose={actionFn}
        data={{
          name: 'test_1',
          hash: '1234abcd',
          action: {
            label: 'Do it',
            url: 'https://www.google.com/'
          },
          dismiss: {
            label: "Don't do it"
          }
        }}
      />
    );
    wrapper.find('button.dismissBtn').simulate('click');
    expect(actionFn).toHaveBeenCalled();
  });
});
