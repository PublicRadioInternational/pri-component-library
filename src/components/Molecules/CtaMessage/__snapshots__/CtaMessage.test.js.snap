// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CtaMessage /> Matches the CTA Message Action button snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  >
    <div
      className="actions"
    >
      <a
        className="btnWhite actionBtn"
        href="https://www.google.com/"
        onClick={[Function]}
      >
        Yes
      </a>
    </div>
  </div>
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message Blue Action button snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  >
    <div
      className="actions"
    >
      <a
        className="btnBlue actionBtn"
        href="https://www.google.com/"
        onClick={[Function]}
      >
        Yes
      </a>
    </div>
  </div>
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message Decription snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  >
    <div
      className="description"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Test description",
        }
      }
    />
  </div>
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message Default snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  />
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message Dismiss button snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  >
    <div
      className="actions"
    >
      <a
        className="btnBlue actionBtn"
        href="https://www.google.com/"
        onClick={[Function]}
      >
        Yes
      </a>
      <button
        aria-expanded={null}
        aria-haspopup={null}
        aria-label={null}
        className="btnDark dismissBtn"
        onClick={[Function]}
        type="button"
      >
        No
      </button>
    </div>
  </div>
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message Logo snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  >
    <div
      className="logo"
    >
      <svg
        aria-hidden={null}
        className=""
        fill="currentcolor"
        height={26}
        version={1.1}
        viewBox="0 0 151 32"
        width={123}
      >
        <title>
          PRI
        </title>
        <path
          d="M47.53 6.44h3.744c-0.129 1.787-1.641 3.204-3.497 3.204h-0.246v1.987h0.246c3.064 0 5.553-2.439 5.553-5.434v-5.436h-5.798v5.678zM6.797 5.436v5.436h5.8v-5.677h-3.751c0.129-1.781 1.641-3.206 3.499-3.206h0.252v-1.989h-0.252c-3.057 0-5.548 2.441-5.548 5.436zM0 5.436v5.436h5.8v-5.677h-3.744c0.129-1.781 1.639-3.206 3.499-3.206h0.246v-1.989h-0.246c-3.059 0-5.555 2.441-5.555 5.436zM54.322 6.44h3.75c-0.129 1.787-1.639 3.204-3.503 3.204h-0.246v1.987h0.246c3.064 0 5.559-2.439 5.559-5.434v-5.436h-5.804v5.678zM36.386 1.555c-8.222-3.412-17.749 0.351-21.239 8.394-3.486 8.048 0.359 17.368 8.586 20.781 3.976 1.658 8.38 1.688 12.393 0.11 4.007-1.591 7.15-4.614 8.838-8.506 3.492-8.041-0.359-17.36-8.578-20.779zM41.549 12.588c0.928-0.106 1.808-0.293 2.618-0.571 0.637 2.091 0.797 4.327 0.409 6.545-1.077 0.207-2.228 0.321-3.423 0.334 0.525-1.892 0.624-4.13 0.396-6.308zM43.68 10.672c-0.713 0.235-1.492 0.398-2.315 0.484-0.278-1.677-0.738-3.268-1.318-4.562 0.261-0.142 0.459-0.304 0.607-0.469 1.298 1.324 2.321 2.87 3.027 4.547zM28.568 30.489c-0.538-0.056-1.070-0.136-1.602-0.258 1.212-1.371 2.286-2.754 3.219-4.114 1.395 0.372 2.786 0.639 4.155 0.781-1.486 1.313-3.382 2.588-5.772 3.591zM18.287 24.8c-1.802-2.343-2.831-5.199-2.939-8.158 0.68 0.922 1.466 1.815 2.345 2.657-0.049 1.649 0.105 3.503 0.594 5.501zM19.179 20.611c1.057 0.857 2.216 1.651 3.46 2.373-0.347 1.597-0.62 3.311-0.794 5.118-0.444-0.295-0.883-0.603-1.292-0.947-0.922-2.381-1.307-4.592-1.374-6.545zM19.209 18.74c0.192-2.93 1.035-5.101 1.516-6.134 1.283 1.113 2.724 2.146 4.285 3.075-0.769 1.748-1.468 3.69-2.024 5.837-1.389-0.838-2.657-1.772-3.778-2.779zM32.84 9.738c-0.693-0.291-1.361-0.618-1.998-0.965 1.055-1.324 2.011-2.297 2.68-2.922 0.26 0.131 0.527 0.261 0.818 0.372 0.291 0.123 0.583 0.23 0.88 0.329 0.013 0.902-0.043 2.254-0.278 3.925-0.71-0.205-1.408-0.452-2.101-0.738zM34.696 11.904c-0.329 1.696-0.84 3.641-1.658 5.749-1.053-0.319-2.106-0.695-3.146-1.126-1.046-0.435-2.050-0.915-3.014-1.443 0.982-2.030 2.056-3.742 3.075-5.156 0.738 0.416 1.505 0.786 2.308 1.128 0.814 0.336 1.623 0.612 2.435 0.848zM29.584 8.018c-1.027-0.668-1.949-1.417-2.743-2.209 1.35-0.898 2.73-1.611 3.962-2.030 0.364 0.444 0.848 0.891 1.43 1.305-0.706 0.672-1.634 1.651-2.65 2.934zM30.103 2.495c-1.343 0.489-2.81 1.27-4.228 2.23-0.514-0.644-0.93-1.3-1.251-1.97 1.701-0.667 3.535-1.023 5.393-1.027-0.013 0.248 0.013 0.504 0.086 0.767zM28.71 9.182c-1.027 1.423-2.104 3.156-3.092 5.187-1.55-0.93-2.96-1.976-4.205-3.096 0.909-1.634 2.482-3.257 4.235-4.607 0.872 0.911 1.907 1.753 3.062 2.515zM26.279 16.399c0.977 0.527 1.992 1.014 3.038 1.447 1.051 0.441 2.112 0.816 3.169 1.141-0.753 1.744-1.714 3.578-2.934 5.464-0.896-0.276-1.789-0.592-2.681-0.956-0.885-0.375-1.746-0.784-2.569-1.223 0.542-2.172 1.221-4.119 1.977-5.873zM31.018 24.854c1.195-1.888 2.138-3.72 2.885-5.471 1.778 0.45 3.537 0.741 5.24 0.87-0.424 1.061-1.438 3.158-3.456 5.331-1.514-0.084-3.089-0.334-4.668-0.73zM34.448 18.042c0.807-2.112 1.331-4.067 1.667-5.79 1.361 0.291 2.702 0.428 3.974 0.416 0.23 2.174 0.129 4.407-0.444 6.188-1.679-0.101-3.426-0.372-5.197-0.814zM36.362 10.834c0.243-1.615 0.31-2.954 0.323-3.925 0.693 0.118 1.356 0.149 1.944 0.093 0.545 1.171 0.982 2.642 1.268 4.229-1.139-0.009-2.329-0.142-3.535-0.398zM39.574 5.133c-0.155 0.217-0.484 0.332-0.687 0.383-0.99 0.235-2.513 0-3.979-0.605-2.347-0.973-3.542-2.428-3.43-3.115 1.46 0.138 2.92 0.489 4.34 1.077 1.4 0.583 2.655 1.348 3.757 2.259zM23.289 3.35c0.364 0.76 0.835 1.511 1.399 2.233-1.752 1.356-3.331 2.978-4.351 4.657-0.831-0.863-1.55-1.74-2.142-2.639 1.361-1.813 3.118-3.255 5.094-4.252zM17.319 8.903c0.652 0.911 1.425 1.796 2.293 2.663-0.433 0.807-1.387 2.876-1.765 5.83-0.954-1.025-1.752-2.11-2.385-3.238 0.179-1.223 0.515-2.45 1.034-3.647 0.241-0.562 0.519-1.096 0.822-1.608zM23.244 28.93c0.149-1.845 0.396-3.585 0.732-5.208 0.753 0.39 1.522 0.751 2.321 1.083 0.805 0.336 1.608 0.626 2.413 0.883-0.958 1.363-2.056 2.747-3.309 4.125-0.372-0.116-0.739-0.252-1.098-0.394-0.366-0.155-0.717-0.319-1.059-0.489zM31.875 30.446c1.808-1.053 3.279-2.231 4.455-3.413 1.245 0.039 2.452-0.037 3.604-0.204-2.248 1.987-5.070 3.244-8.059 3.617zM41.648 25.042c-1.257 0.332-2.612 0.517-4.028 0.57 1.901-2.315 2.76-4.412 3.062-5.288 1.225 0.015 2.416-0.069 3.548-0.241-0.168 0.571-0.364 1.132-0.612 1.69-0.515 1.201-1.184 2.291-1.97 3.27z"
          fill="currentColor"
        />
        <path
          d="M126.74 30.947h10.504v-30.030h-10.504v30.030zM96.642 10.644c0 3.051-1.281 5.805-3.509 7.563-2.681 2.125-6.099 2.373-8.433 2.373h-3.813v10.369h-10.554v-30.032h12.666c3.081 0 6.313 0.121 8.931 1.602 2.948 1.71 4.711 4.749 4.711 8.125zM86.255 10.933c0-1.529-0.855-2.024-3.516-2.024h-1.85v4.091h1.595c2.504 0 3.772-0.321 3.772-2.067zM120.023 18.059c0.444 1.051 5.458 12.888 5.458 12.888h-11.581c0 0-3.918-10.061-4.257-10.914h-1.554v10.914h-10.554v-30.030h15.599c3.425 0 6.016 0.612 7.936 1.865 2.526 1.557 3.921 4.209 3.921 7.439-0.002 3.652-1.638 6.14-4.969 7.837zM114.178 10.431c0-0.725-0.383-1.369-0.99-1.684-0.612-0.297-1.43-0.297-2.228-0.297h-2.872v4.005h2.538c2.952 0 3.552-0.803 3.552-2.024z"
          fill="currentColor"
        />
        <path
          d="M145.954 0.917c-2.644 0-4.687 2.011-4.687 4.594 0 2.573 2.043 4.592 4.687 4.592 2.648 0 4.698-2.019 4.698-4.592 0-2.582-2.050-4.594-4.698-4.594zM145.954 1.569c2.273 0 3.968 1.744 3.968 3.942 0 2.19-1.696 3.936-3.968 3.936-2.267 0-3.962-1.746-3.962-3.936 0-2.198 1.696-3.942 3.962-3.942zM144.92 5.8h1.128l1.52 2.36h0.788l-1.613-2.36c0.73-0.116 1.361-0.538 1.361-1.471 0-0.954-0.545-1.475-1.763-1.475h-2.088v5.307h0.668v-2.36zM144.92 5.249v-1.836h1.225c0.657 0 1.287 0.174 1.287 0.915 0 0.898-0.812 0.921-1.628 0.921h-0.883z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message Title snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  >
    <h3
      className="title"
    >
      Test Title
    </h3>
  </div>
</aside>
`;

exports[`<CtaMessage /> Matches the CTA Message w/o buttons snapshot 1`] = `
<aside
  className="pushDown"
>
  <button
    className="close"
    onClick={[Function]}
  >
    <svg
      aria-hidden={null}
      aria-label="Close"
      aria-labelledby={null}
      className="svg"
      fill="currentcolor"
      height={18}
      version={1.1}
      viewBox="0 0 28 28"
      width={18}
    >
      <path
        d="M27.745 22.495c-0-0-0-0-0-0l-8.494-8.494 8.494-8.494c0-0 0-0 0-0 0.091-0.091 0.158-0.198 0.2-0.312 0.116-0.311 0.050-0.675-0.2-0.925l-4.013-4.013c-0.25-0.25-0.614-0.316-0.925-0.2-0.114 0.042-0.221 0.109-0.312 0.2 0 0-0 0-0 0l-8.494 8.494-8.494-8.494c-0-0-0-0-0-0-0.091-0.091-0.198-0.158-0.312-0.2-0.311-0.116-0.675-0.050-0.925 0.2l-4.013 4.013c-0.25 0.25-0.316 0.614-0.2 0.925 0.042 0.114 0.109 0.221 0.2 0.312 0 0 0 0 0 0l8.494 8.494-8.494 8.494c-0 0-0 0-0 0-0.091 0.091-0.157 0.198-0.2 0.312-0.116 0.311-0.050 0.675 0.2 0.925l4.013 4.013c0.25 0.25 0.614 0.316 0.925 0.2 0.114-0.042 0.221-0.109 0.312-0.2 0-0 0-0 0-0l8.494-8.494 8.494 8.494c0 0 0 0 0 0 0.092 0.091 0.198 0.158 0.312 0.2 0.311 0.116 0.675 0.050 0.925-0.2l4.013-4.013c0.25-0.25 0.316-0.614 0.2-0.925-0.042-0.114-0.109-0.221-0.2-0.312z"
      />
    </svg>
  </button>
  <div
    className="content"
  />
</aside>
`;
